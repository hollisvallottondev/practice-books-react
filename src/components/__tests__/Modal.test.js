import React from 'react';
import Modal from '../Modal/index.jsx';
import {ConfirmDelete, EditBook} from '../BookCard/index.jsx';
import booksMocks from '../../mocks/books_seed.json';
import renderer from 'react-test-renderer';

test('Modal renders edit correctly', () => {
	const handleModalClose = jest.fn(() => console.log('Handle modal close'));
	const handleBookSave = jest.fn(() => console.log('Handle book save'));
	const index = 0;
	const book = booksMocks[index];
	const openModal = true;

	const component = renderer.create(
		<Modal 
			open = {openModal}
			 handleClose = {handleModalClose}>
			<EditBook 
				book ={book} 		
				handleCancel={handleModalClose} 					
				handleBookSave={handleBookSave}/>
		</Modal>
	)

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Modal renders confirm correctly', () => {
	const handleModalClose = jest.fn(() => console.log('Handle modal close'));
	const handleConfirmDelete = jest.fn(() => console.log('Handle confirm delete'));
	const index = 0;
	const book = booksMocks[index];
	const openModal = true;

	const component = renderer.create(
		<Modal 
			open = {openModal}
			 handleClose = {handleModalClose}>
			<ConfirmDelete 
				focusDeleteIndex = {index} 
				book={book} 
				handleModalClose={handleModalClose} 
				handleConfirmDelete={handleConfirmDelete}/>
		</Modal>
	)

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});


