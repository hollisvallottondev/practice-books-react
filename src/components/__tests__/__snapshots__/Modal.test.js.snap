// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders confirm correctly 1`] = `
<div>
  <div>
    <div>
      <i
        className="fas fa-times-circle"
        onClick={[MockFunction]}
      />
      <div>
        <p>
          Are you sure you want to delete Girls Just Want to Have Fun by Everett Chotty?
        </p>
        <div>
          <button
            onClick={[Function]}
          >
            Cancel
          </button>
          <button
            onClick={[Function]}
          >
            Confirm
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal renders edit correctly 1`] = `
<div>
  <div>
    <div>
      <i
        className="fas fa-times-circle"
        onClick={[MockFunction]}
      />
      <div>
        <div
          className="undefined "
          onClick={[Function]}
        >
          <div>
            <div>
              <h3 />
            </div>
            <div />
          </div>
          <div>
            <button
              onClick={[MockFunction]}
            >
              Cancel
            </button>
            <button
              onClick={[Function]}
            >
              Save
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
